### ISSUE TITLE
Class imbalance - Need to set class weights

### ISSUE THREAD

**Issue by sequoiagrove** [qubvel/segmentation_models.pytorch#480]
Hi,
Im trying to move from CNTK to Pytorch.
I have 3 classes: one very large in the image, one 1/5 smaller and a very small dot 1000 times smaller than the large.
CNTK UNET learned it fine because I used a Dice loss with class weights.  [1,200,1000]. 
In this library it quickly learns the too first classes. in like 8 epochs. IOU = 0.95 It's very impressive, althoguh I'd assume it should be lower than that when an entire class is ignored totally if it is the mean of each class IOU and not the mean of pixel accuracy (to which a small dot would have barely any effect).  but the 3rd class is always all zeros. I tried training 500 epochs but the 3rd class is always ignored. I also tried Jaccuard loss. same result.
I've searched for the concept of class weights in this repo but it doesnt seem to have it?

**Comment by caiya55** [comment]
you can use pytorch_toolbelt and its joint loss function to achieve this. My code works like this:
'''
from pytorch_toolbelt import losses as L
def ce_Loss(score_map, labels):
    labels = labels[:,0,:,:]
    weight_CE = torch.FloatTensor([1,1,1,1,5,1,1,1]).cuda()
    cross_loss = nn.CrossEntropyLoss(weight=weight_CE)
    loss = cross_loss(score_map, labels)
    return loss

loss = L.JointLoss(ce_Loss, ce_Loss, 0.5, 0.5)
loss.__name__ = 'focal_jaccard'
'''

**Comment by davodogster** [comment]
@caiya55 Hi I installed this and it uninstalled pytorch 1.7.1 and installed pytorch 1.9 which was not good. I need to use PyT 1.7 with my GPU.

**Comment by caiya55** [comment]
hi，you need to install the older veraion. The newest version will  corrupt your toech environment. Good luck.

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
